frontend/public/components/network-policies/network-policy-form.tsx

```diff
@@ -0,0 +1,354 @@
+    );
+  };
+
+  const isValid = () => {
+    return (
+      networkPolicy.name &&
+      networkPolicy.namespace &&
+      areValidRules(networkPolicy.ingress) &&
+      areValidRules(networkPolicy.egress)
+    );
+  };
+
+  const save = (event) => {
+    event.preventDefault();
+
+    const policy = networkPolicyToK8sResource(networkPolicy);
+    if (isNetworkPolicyConversionError(policy)) {
+      setError(policy.error);
+      return;
+    }
+
+    setInProgress(true);
+    k8sCreate(NetworkPolicyModel, policy).then(
+      () => {
+        setInProgress(false);
+        history.push(
+          resourcePathFromModel(NetworkPolicyModel, networkPolicy.name, networkPolicy.namespace),
+        );
+      },
+      (err) => {
+        setError(err.message);
+        setInProgress(false);
+      },
+    );
+  };
+
+  return (
+    <Form onSubmit={save} className="co-create-networkpolicy">
+      <div className="form-group co-create-networkpolicy__name">
+        <label className="co-required" htmlFor="name">
+          {t('public~Policy name')}
+        </label>
+        <input
+          className="pf-c-form-control"
+          type="text"
+          onChange={handleNameChange}
+          value={networkPolicy.name}
+          placeholder="my-policy"
+          id="name"
+          name="name"
+          required
+        />
+      </div>
+      <div className="form-group co-create-networkpolicy__namespace">
+        <label className="co-required" htmlFor="namespace">
+          {t('public~Policy namespace')}
+        </label>
+        <input
+          className="pf-c-form-control"
+          type="text"
+          onChange={handleNamespaceChange}
+          value={networkPolicy.namespace}
+          id="namespace"
+          name="namespace"
+          aria-describedby="namespace-help"
+          required
+        />
+        <div className="help-block" id="namespace-help">
+          <p>{t('public~The namespace containing pods on which this network policy applies.')}</p>
+        </div>
+      </div>
+      <div className="form-group co-create-networkpolicy__podselector">
+        <NetworkPolicyConditionalSelector
+          selectorType="pod"
+          helpText={t(
+            'public~If no pod selector is provided, the policy will apply to all pods in the namespace.',
+          )}
+          values={networkPolicy.podSelector}
+          onChange={handleMainPodSelectorChange}
+        />
+      </div>
+      <div className="form-group co-create-networkpolicy__type">
+        <Title headingLevel="h2">{t('public~Policy type')}</Title>
+      </div>
+      {showSDNAlert && (
+        <Alert
+          variant="info"
+          title={t(
+            'public~When using the OpenShift SDN cluster network provider, egress network policy is not supported.',
+          )}
+          actionClose={<AlertActionCloseButton onClose={() => setShowSDNAlert(false)} />}
+        />
```

Comment:
<img width="632" alt="Screen Shot 2021-06-30 at 12 44 10 PM" src="https://user-images.githubusercontent.com/1874151/124002386-abca7180-d9a3-11eb-9703-576908af3969.png">

Seems this should be a [warning alert](https://www.patternfly.org/v4/components/alert#inline-types)?
In other forms we display `<Alert>` inline so they aren't floating.

```suggestion
        <Alert
          variant="warning"
          isInline
          className="co-alert"
          title={t(
            'public~When using the OpenShift SDN cluster network provider, egress network policy is not supported.',
          )}
          actionClose={<AlertActionCloseButton onClose={() => setShowSDNAlert(false)} />}
        />
```

Reply:
I was using a warning at first but changed to info. The reason is that this banner is always displayed, even when there's no problem because the user uses ovn-k cni, and we don't want to sound too aggressive or worrying while there are chances that there's no issue at all. Also, having it as a floating box allows the user to dismiss it if desired (if it's perceived as annoying).

This situation isn't perfect for sure, but it's temporary, there's a task with the SDN team to expose information about the CNI plugin in use; when it's done, we will be able to remove this message and adapt the display to whatever is relevant.

Reply:
BTW I could add something to the message like "If your network provider is OVN-Kubernetes, please ignore this message"

Reply:
I agree it should be an info message for the reasons you state.

We could try to get the configuration for users who have authority, but if the SDN team is working to expose it to normal users, I'm fine leaving it as-is.